src/code/irqmgr.c

```diff
@@ -1,122 +1,174 @@
-        osSendMesg(c->queue, (OSMesg) & this->nmiMsg, OS_MESG_NOBLOCK);
+    if (this->resetStatus >= IRQ_RESET_NMI) {
+        osSendMesg(client->queue, (OSMesg) & this->nmiMsg, OS_MESG_NOBLOCK);
     }
 }
 
-void IrqMgr_RemoveClient(IrqMgr* this, IrqMgrClient* c) {
-    IrqMgrClient* iter = this->clients;
-    IrqMgrClient* lastIter = NULL;
+void IrqMgr_RemoveClient(IrqMgr* this, IrqMgrClient* client) {
+    IrqMgrClient* iterClient = this->clients;
+    IrqMgrClient* lastClient = NULL;
     OSIntMask prevInt;
 
     LogUtils_CheckNullPointer("this", this, "../irqmgr.c", 129);
-    LogUtils_CheckNullPointer("c", c, "../irqmgr.c", 130);
+    LogUtils_CheckNullPointer("c", client, "../irqmgr.c", 130);
 
+    // Disable interrupts to prevent a thread context switch while the linked list is modified
     prevInt = osSetIntMask(OS_IM_NONE);
 
-    while (iter != NULL) {
-        if (iter == c) {
-            if (lastIter) {
-                lastIter->prev = c->prev;
+    // Remove the client
+    while (iterClient != NULL) {
+        if (iterClient == client) {
+            if (lastClient != NULL) {
+                lastClient->prev = client->prev;
             } else {
-                this->clients = c->prev;
+                this->clients = client->prev;
             }
             break;
         }
-        lastIter = iter;
-        iter = iter->prev;
+        lastClient = iterClient;
+        iterClient = iterClient->prev;
     }
 
     osSetIntMask(prevInt);
 }
 
-void IrqMgr_SendMesgForClient(IrqMgr* this, OSMesg msg) {
-    IrqMgrClient* iter = this->clients;
+/**
+ * Send `msg` to every registered client if the message queue is not full. The message is
+ * appended to the back of the queue.
+ */
+void IrqMgr_SendMesgToClients(IrqMgr* this, OSMesg msg) {
+    IrqMgrClient* client;
 
-    while (iter != NULL) {
-        if (iter->queue->validCount >= iter->queue->msgCount) {
+    for (client = this->clients; client != NULL; client = client->prev) {
+        if (MQ_IS_FULL(client->queue)) {
             // "irqmgr_SendMesgForClient: Message queue is overflowing mq=%08x cnt=%d"
             osSyncPrintf(
                 VT_COL(RED, WHITE) "irqmgr_SendMesgForClient:メッセージキューがあふれています mq=%08x cnt=%d\n" VT_RST,
-                iter->queue, iter->queue->validCount);
+                client->queue, MQ_GET_COUNT(client->queue));
         } else {
-            osSendMesg(iter->queue, msg, OS_MESG_NOBLOCK);
+            osSendMesg(client->queue, msg, OS_MESG_NOBLOCK);
         }
-
-        iter = iter->prev;
     }
 }
 
-void IrqMgr_JamMesgForClient(IrqMgr* this, OSMesg msg) {
-    IrqMgrClient* iter = this->clients;
-
-    while (iter != NULL) {
-        if (iter->queue->validCount >= iter->queue->msgCount) {
+/**
+ * Send `msg` to every registered client if the message queue is not full. This appears to be for
+ * high-priority messages that should be jammed to the front of the queue, however a bug prevents
```

Comment:
Is this a technical sense of "jam"?

Reply:
In the libultra message queue sense, to "jam" a message means to add it to the front of the message queue rather than the back.